### PR TITLE
Add tutorial for configuring automatic deployment

### PR DESCRIPTION
**Pull request by RukshiW**

### CODE REVIEW

**Comment by RukshiW** [comment]
Fixes https://github.com/wso2-enterprise/choreo/issues/15683

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+12. Click the refresh icon next to **Build and Deployment History**.
+
+     ![Refresh build and deployment history](../assets/img/tutorials/deploy-automatically/refresh-build-and-deployment-history.png){.cInlineImage-threeQuarters}
+
+    The **Build and Deployment History** section will display a new record with the **Automatic** label as shown below for the automatic build you triggered with your last commit.
+
+    ![Automatic deployment record](../assets/img/tutorials/deploy-automatically/automatic-deployment-record.png){.cInlineImage-threeQuarters}
+
+    You can click on this record to view the commit that triggered it in GitHub.
```
Comment (Drifftr) [inline en/docs/tutorials/configure-automatic-deployment.md:98]: If you click on the record it will open the build info pane. If you click on the commit hash it will open the commit that triggered the build in github.

**Review by Drifftr — DISMISSED** [review]

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Configure Automatic Deployment
+
+When you develop a Choreo component, you need to commit its source code into a GitHub repository. You can select a repository that Choreo manages or a repository that you own and manage.
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:3]: You can't select a repository that Choreo manages right? Should we rephrase it as "You can choose to commit it to a Choreo managed GitHub repository or a repository that you own and manage"

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+Choreo-managed repositories are not accessible to external users. Therefore, when multiple developers need to develop components in collaboration, they should connect their own repositories to save the source code. In such organizations, deploying a component manually each time a developer updates the component design can be cumbersome. Therefore, Choreo allows you to configure automatic deployment for your components. When you enable automatic deployment, every commit that changes the component implementation automatically deploys the component to the development environment and triggers a build.
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:5]: do we need the word "own"? Can we rephrase it to "Therefore, when multiple developers need to develop components in collaboration, they should connect their repositories to save the source code."

Comment (shaniR) [reply en/docs/tutorials/configure-automatic-deployment.md:5]: Can we rephrase this to "In such organizations, manually deploying a component each time a developer updates the component design can be cumbersome"

Comment (shaniR) [reply en/docs/tutorials/configure-automatic-deployment.md:5]: is it every commit? or every commit with a diff specifically?

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+This tutorial walks you through the steps to enable automatic deployment for a component. In this tutorial, you will:
+
+- Configure automatic deployment for a REST API.
+- Commit updates to the REST API implementation and observe automatic builds.
+
+## Step 1: Connect your own GitHub repository for a REST API
+
+Let's first create a REST API connecting a GitHub repository you own to save the implementation.
+
+For instructions, see [Connect Your Own GitHub Repository to Choreo](connect-your-own-github-repository-to-choreo.md)
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:16]: can we leave out "own"?

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+    !!! info
+        Choreo requires you to perform the first deployment of each component manually so that you can provide values for any configurable variables that the implementation may include.
+
+2. Click the **Automatic Deployment** toggle to enable automatic deployment.
+
+    !!! info
+        You need to verify your GitHub account when you enable automatic deployment for the first time.
+
+3. In the message that appears on the bottom-right of the page to configure an alert for build failures, click **Configure Alert**.
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:32]: Shall we rephrase this as "In the message that appears on the bottom-right of the page , click **Configure Alert** to configure an alert for build failures."

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 3 above)
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Rewrote this paragraph.

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+1. In the Choreo Console, navigate back to the REST API component for which you enabled automatic deployment.
+
+2. To update the implementation, click **Overview** in the left pane and then click **Edit Code**. The Web Editor opens.
+
+3. In the Web Editor, make an edit to the REST API implementation. For example, click the last return statement, replace `Hello` in the statement with `Hi`, and click **Save**.
+
+    !!! info
+        To edit this value, double-click it.
+
+    ![Edit implementation](../assets/img/tutorials/deploy-automatically/edit-implementation.textClipping){.cInlineImage-full}
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:56]: This image is not rendering correctly

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
+# Configure Automatic Deployment
+
```
Comment (shaniR) [inline en/docs/tutorials/configure-automatic-deployment.md:2]: Can we check if we can convert the full sized images to three quarters or half?

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+3. In the message that appears on the bottom-right of the page , click **Configure Alert*. The **Alert Configuration** tab of the **Settings** page opens.
+
+4. For the **Component build and deployment failure alert**, enter a valid email address and click the toggle to enable the alert.
+
+    ![Enable build failure alerts](../assets/img/tutorials/deploy-automatically/enable-build-failure-alerts.png){.cInlineImage-threeQuarters}
+
+Now automatic deployment is enabled for your REST API and a build should be automatically triggered every time you commit a change to the REST API implementation. To verify this, move on to Step 3.
+
+## Step 3: Commit changes and trigger the automatic build
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Changed this to " Step 3: Commit changes to trigger the automatic build"

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+2. To update the implementation, click **Overview** in the left pane and then click **Edit Code**. The Web Editor opens.
+
+3. In the Web Editor, make an edit to the REST API implementation. For example, click the last return statement, replace `Hello` in the statement with `Hi`, and click **Save**.
+
+    !!! info
+        To edit this value, double-click it.
+
+    ![Edit implementation](../assets/img/tutorials/deploy-automatically/edit-implementation.textClipping){.cInlineImage-full}
+
+4. Click **Sync with Choreo Upstream** in the Git Status Bar.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: There is no specific standard mentioned in [1]. Changed it to lower case.

[1] https://docs.google.com/document/d/1C6t7fRbr2rXLzvdrv1wZLJkfqFY8MWR1SAzsUWoNtj0/edit

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 6 above)
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Fixed the image link.

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    You will get a message prompting you to configure your `user.name` and `user.email`. To proceed, follow the steps given below:
+
+    1. In the message, click **Open Git Log**.
+   
+    2. Click **Terminal** to open the in-built terminal and issue the following commands in it.
+   
+        - `git config --global user.email "<YOUR_GITHUB_EMAIL_ADDRESS>"`
+        - `git config --global user.name "<YOUR_GITHUB_USERNAME"`
+
+6. To push the changes to the private GitHub repository maintained by Choreo, click **0↓ 1↑** in the Git Status Bar.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Changed this text.

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    !!! info
+        You need to verify your GitHub account when you enable automatic deployment for the first time.
+
+3. In the message that appears on the bottom-right of the page to configure an alert for build failures, click **Configure Alert**.
+
+4. For the **Component build and deployment failure alert**, enter a valid email address and click the toggle to enable the alert.
+
+    ![Enable build failure alerts](../assets/img/tutorials/deploy-automatically/enable-build-failure-alerts.png){.cInlineImage-threeQuarters}
+
+Now automatic deployment is enabled for your REST API and a build should be automatically triggered every time you commit a change to the REST API implementation. To verify this, move on to Step 3.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md:41]: Even committing edits to readme.md file triggers automatic builds. Tried it.

@Drifftr can you please confirm?

File: en/docs/tutorials/configure-automatic-deployment.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+5. In the left pane, enter a commit message (e.g., `Implement REST API`) and click the tick.
+
+    ![Commit message](../assets/img/tutorials/rest-api/commit-message.png){.cInlineImage-full}
+
+    You will get a message prompting you to configure your `user.name` and `user.email`. To proceed, follow the steps given below:
+
+    1. In the message, click **Open Git Log**.
+   
+    2. Click **Terminal** to open the in-built terminal and issue the following commands in it.
```
[the remark this answers is not in the document]
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Updated as suggested.

However, it seems like the placeholder text makes it clear enough and specially mentioning it makes the step a bit wordier than necessary. What do you think?

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 7 above)
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md:2]: Changed to three quarters

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Drifftr) [reply en/docs/tutorials/configure-automatic-deployment.md:41]: This triggers with a commit to any file in the repo, it doesn't have to be the specific bal file. Committing without a diff is also possible but it's not straightforward(Have to use special switches for that) and very rare. However even a commit without a diff, triggers a build.

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 2 above)
Comment (RukshiW) [reply en/docs/tutorials/configure-automatic-deployment.md]: Addressed

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/tutorials/configure-automatic-deployment.md:41]: cool, thanks for verifying.

**Review by shaniR — DISMISSED** [review]

File: en/docs/tutorials/configure-automatic-deployment.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (shaniR) [reply en/docs/tutorials/configure-automatic-deployment.md]: It does, but sometimes you can miss it when trying this out. Better to mention it to avoid any steps being missed.

**Review by shaniR — APPROVED** [review]
